apis/v1alpha1/gateway_types.go

```diff
@@ -731,10 +671,10 @@ const (
 	// interoperability.
 	ListenerConditionConflicted ListenerConditionType = "Conflicted"
 
-	// ListenerReasonHostnameConflict is used when the Listener
-	// violates the Hostname match constraints that allow collapsing
-	// Listeners. For example, this reason would be used when multiple
-	// Listeners on the same port use the "Any" hostname match type.
+	// ListenerReasonHostnameConflict is used when the Listener conflicts with
+	// hostnames in other Gateway Listeners. For example, this reason would be
```

Comment:
nit: `s/other Gateway Listeners/other Listeners` because Listener, although not a resource, is a well known construct. This made me think and confuse about  listeners on other Gateway. (that's probably because merging is a topic on our minds right now).

Reply:
Updated, thanks!